src/coordination/coordinator_state_machine.cpp

```diff
@@ -181,47 +181,60 @@ auto CoordinatorStateMachine::SerializeUpdateUUIDForInstance(InstanceUUIDUpdate
   return CreateLog({{"action", RaftLogAction::UPDATE_UUID_FOR_INSTANCE}, {"info", instance_uuid_change}});
 }
 
-auto CoordinatorStateMachine::DecodeLog(buffer &data) -> std::pair<TRaftLog, RaftLogAction> {
+auto CoordinatorStateMachine::DecodeLog(buffer &data) -> std::optional<std::pair<TRaftLog, RaftLogAction>> {
   buffer_serializer bs(data);
-  auto const json = nlohmann::json::parse(bs.get_str());
+  std::string const log_str = bs.get_str();
+  if (log_str.empty()) {
+    return std::nullopt;  // this is returned only when the log is configuration log
+  }
+  auto const json = nlohmann::json::parse(log_str);
   auto const action = json["action"].get<RaftLogAction>();
   auto const &info = json.at("info");
 
   switch (action) {
     case RaftLogAction::OPEN_LOCK:
       [[fallthrough]];
     case RaftLogAction::CLOSE_LOCK: {
-      return {std::monostate{}, action};
+      return std::pair{std::monostate{}, action};
     }
     case RaftLogAction::REGISTER_REPLICATION_INSTANCE:
-      return {info.get<CoordinatorToReplicaConfig>(), action};
+      return std::pair{info.get<CoordinatorToReplicaConfig>(), action};
     case RaftLogAction::UPDATE_UUID_OF_NEW_MAIN:
-      return {info.get<utils::UUID>(), action};
+      return std::pair{info.get<utils::UUID>(), action};
     case RaftLogAction::UPDATE_UUID_FOR_INSTANCE:
     case RaftLogAction::SET_INSTANCE_AS_MAIN:
-      return {info.get<InstanceUUIDUpdate>(), action};
+      return std::pair{info.get<InstanceUUIDUpdate>(), action};
     case RaftLogAction::UNREGISTER_REPLICATION_INSTANCE:
     case RaftLogAction::INSTANCE_NEEDS_DEMOTE:
       [[fallthrough]];
     case RaftLogAction::SET_INSTANCE_AS_REPLICA:
-      return {info.get<std::string>(), action};
+      return std::pair{info.get<std::string>(), action};
   }
   throw std::runtime_error("Unknown action");
 }
 
 auto CoordinatorStateMachine::pre_commit(ulong const /*log_idx*/, buffer & /*data*/) -> ptr<buffer> { return nullptr; }
 
 auto CoordinatorStateMachine::commit(ulong const log_idx, buffer &data) -> ptr<buffer> {
+  auto const return_successful_commit = [this, &log_idx]() {
+    last_committed_idx_ = log_idx;
+    logger_.Log(nuraft_log_level::TRACE, fmt::format("Last commit index: {}", last_committed_idx_));
+    ptr<buffer> ret = buffer::alloc(sizeof(log_idx));
+    buffer_serializer bs_ret(ret);
+    bs_ret.put_u64(log_idx);
+    return ret;
+  };
```

Comment:
why this change, in both cases (if-else) we do the same?

Reply:
You are right, this can be same part without func, I thought just of isolating same part in function